uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/instances/leave-request-instance_test.js

```diff
@@ -587,5 +529,80 @@ define([
         }).value;
       }
     });
+
+    describe('roleOf()', function () {
+
+      var promise,
+        defer;
+
+      afterEach(function () {
+        $rootScope.$apply();
+      });
+
+      describe('when contact_id of leave request is same as contact id of parameter', function () {
+
+        beforeEach(function () {
+          spyOn(LeaveRequestAPI, 'isManagedBy').and.callThrough();
+          //dummy contact id
+          LeaveRequestInstance.contact_id = "101";
+          promise = LeaveRequestInstance.roleOf({
+            id: '101'
+          });
+        });
+
+        it('returns owner', function () {
+          promise.then(function (result) {
+            expect(result).toBe('owner');
+          });
+        })
+      });
+
+      describe('when contact_id of leave request is not same as contact id of parameter', function () {
+
+        describe('when isManagedBy return true', function () {
+
+          beforeEach(function () {
+            spyOn(LeaveRequestAPI, 'isManagedBy').and.callFake(function () {
+              defer = $q.defer();
+              defer.resolve(true);
+              return defer.promise;
+            });
+            commonSetup();
```

Comment:
Please stop obfuscating the meaningful (as in, that give the context of the test) setup of tests into generically-named functions like `commonSetup`. Either you give them a meaningful name or (preferred) just do the setup in the `beforeEach` directly if it's not too complicated, so that you make it explicit and make the tests more understandable without the need to look for this setup function somewhere else

Also, in this particular case this function makes things more confusing because you're not using it for the "contact as owner" case